### PR TITLE
feat(ui): Refactor/fix group sidebar headers

### PR DESCRIPTION
**Pull request by billyvg**
* Refactor our group sidebar headers into a new component `<SidebarSection>`. This handles the header as well as the content so that we can have more consistent spacing between the sidebar sections.
* Fixes the missing header border. Something must have changed with browser rendering, as the CSS was still there.
* Update to use `requestPromise`
* Removed usage of `<LoadingIndicator>` in favor of `<Placeholder>`

image

### CODE REVIEW

**Comment by billyvg** [comment]
Fixed external issues:
image

**Review by matejminar — APPROVED** [review]
LGTM

**Comment by priscilawebdev** [comment]
it looks good to me, but I wouldn't decrease the margin between the Headers and the Content
 
image

**Review by priscilawebdev — APPROVED** [review]

File: src/sentry/static/sentry/app/components/group/sidebar.tsx [hunk 1]
```diff
@@ -134,20 +139,9 @@ class GroupSidebar extends React.Component<Props, State> {
...
-              clearIndicators();
-            },
-            error: () => {
-              this.setState({error: true});
-              clearIndicators();
-            },
-          });
+        complete: async () => {
+          await this.fetchParticipants();
+          clearIndicators();
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/group/sidebar.tsx:144]: Do you need a `catch` here to clear indicators?

**Comment by billyvg** [comment]
@priscilawebdev I wanted the blocks to be closer to the header so you get a better sense of the sections, but maybe having the border will help w/ that, I'll add back the margins and we can adjust as needed

File: src/sentry/static/sentry/app/components/group/sidebar.tsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/group/sidebar.tsx:144]: @markstory Nope, there's a try/catch in `fetchParticipants`, so this should never reject.

**Pull request merged**